src/ExtraValuesProvider.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Wikibase\DataModel;
+
+/**
+ * Provides extra values for the serialization of an object.
+ *
+ * @since 2.1
+ *
+ * @licence GNU GPL v2+
+ * @author Bene* < benestar.wikimedia@gmail.com >
+ */
+interface ExtraValuesProvider {
+
+	/**
+	 * @param mixed $object
+	 * @return bool
+	 */
+	public function isProviderFor( $object );
+
+	/**
+	 * @param mixed $object
+	 * @return array
```

Comment:
This interface looks surprisingly similar to the DispatchableSerializer interface. Basically the two concepts are very similar. The only difference is that the DispatchingSerializer chooses one serializer and uses only that while this implementation combines and merges all serializations.